### ISSUE TITLE
[Bug] Dependency resolution

### ISSUE THREAD

**Issue by Sobes76rus** [David-OConnor/pyflow#117]
Hello there!
I'm trying to install `black` + `uvicorn`
They have `click` as dependency
`black`:  `click>=7.1.2`
`uvicorn`: `click>=7.*`

As u can see, the right version for click is at least `7.1.2`
But `pyflow` install two versions: `8.0.1` for `black` and `7.0.0` for `uvicorn`

After that `pyflow` throw an error
```
Can't find a compatible package for Req { name: "click", constraints: [Constraint { type_: Exact, version: 8.0.1 }], extra: None, sys_platform: None, python_version: None, install_with_extras: None, path: None, git: None }```

**Comment by Sobes76rus** [comment]
On `pypi.org` `click` verison for `uvicorn` is `click (==7.*)` 
But `pyflow` resolves it as `==7.0`
```
Req { name: "click", constraints: [Constraint { type_: Exact, version: 7.0.0 }], extra: None, sys_platform: None, python_version: None, install_with_extras: None, path: None, git: None }
```

**Comment by David-OConnor** [comment]
Thanks. Fixing this sort of bug is top priority.
